Adding dbt MCP documentation (#7743)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-ai/setup-local-mcp.md

---
title: "Set Up local MCP"
sidebar_label: "Set Up local MCP"
description: "Learn how to set up the local dbt-mcp server"
id: "setup-local-mcp"
---

# Set up local MCP <Lifecycle status="beta" />

> [The local dbt MCP server](https://github.com/dbt-labs/dbt-mcp) can be hosted locally on your machine or virtually. To set up the local dbt-mcp server, follow these directions.

Reviewer: ```suggestion
[The local dbt MCP server](https://github.com/dbt-labs/dbt-mcp) can be hosted locally on your machine or virtually. To set up the local dbt MCP server, follow these directions.
```

> 1. [Install uv](https://docs.astral.sh/uv/getting-started/installation/) to install dbt-mcp and [related dependencies](https://github.com/dbt-labs/dbt-mcp/blob/main/pyproject.toml) into an isolated virtual environment. 

Reviewer: ```suggestion
1. [Install uv](https://docs.astral.sh/uv/getting-started/installation/) to install `dbt-mcp` and [related dependencies](https://github.com/dbt-labs/dbt-mcp/blob/main/pyproject.toml) into an isolated virtual environment. 
```

2. Create an `.env` file to set your environment variables. 

Here is an example of the file:

```code
DBT_HOST=cloud.getdbt.com
DBT_PROD_ENV_ID=your-production-environment-id
DBT_DEV_ENV_ID=your-development-environment-id
DBT_USER_ID=your-user-id
DBT_TOKEN=your-service-token
DBT_PROJECT_DIR=/path/to/your/dbt/project
DBT_PATH=/path/to/your/dbt/executable
MULTICELL_ACCOUNT_PREFIX=your-account-prefix
```
 You will need this file for integrating with MCP-compatible tools. 
 Here are the environment variables you could supply:

### Setting Environment Variables

You will need to configure environment variables to access the tools. If you are only using the dbt CLI commands, you do not need to supply the dbt platform specific environment variables and vice versa. 

#### Configuration for Discovery, Semantic Layer, and SQL Tools

| Environment Variable | Required | Description |
| --- | --- | --- |
| DBT_HOST | Required | Your dbt Platform instance hostname. For more information, click [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses). If you are using Multi-cell, don’t include the `ACCOUNT_PREFIX` here. Default is `cloud.getdbt.com`  |

> | MULTICELL_ACCOUNT_PREFIX | Only required for Multi-cell instances | Set your Multi-cell  `ACCOUNT_PREFIX`. If you are not using Multi-cell, do not set. You can learn more [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses).  |

Reviewer: ```suggestion
| MULTICELL_ACCOUNT_PREFIX | Only required for Multi-cell instances | Set your Multi-cell  `ACCOUNT_PREFIX`. If you are not using Multi-cell, don't set this value. You can learn more [about regions and hosting here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses).  |
```

> | MULTICELL_ACCOUNT_PREFIX | Only required for Multi-cell instances | Set your Multi-cell  `ACCwhOUNT_PREFIX`. If you are not using Multi-cell, do not set. You can learn more [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses).  |

Reviewer: ```suggestion
| MULTICELL_ACCOUNT_PREFIX | Only required for Multi-cell instances | Set your Multi-cell  `ACCOUNT_PREFIX`. If you are not using Multi-cell, do not set. You can learn more [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses).  |
```

---
title: "Set Up local MCP"
sidebar_label: "Set Up local MCP"
description: "Learn how to set up the local dbt-mcp server"
id: "setup-local-mcp"
---

# Set up local MCP <Lifecycle status="beta" />

[The local dbt MCP server](https://github.com/dbt-labs/dbt-mcp) can be hosted locally on your machine or virtually. To set up the local dbt MCP server, follow these directions.

1. [Install uv](https://docs.astral.sh/uv/getting-started/installation/) to install `dbt-mcp` and [related dependencies](https://github.com/dbt-labs/dbt-mcp/blob/main/pyproject.toml) into an isolated virtual environment. 

2. Create an `.env` file to set your environment variables. 

Here is an example of the file:

```code
DBT_HOST=cloud.getdbt.com
DBT_PROD_ENV_ID=your-production-environment-id
DBT_DEV_ENV_ID=your-development-environment-id
DBT_USER_ID=your-user-id
DBT_TOKEN=your-service-token
DBT_PROJECT_DIR=/path/to/your/dbt/project
DBT_PATH=/path/to/your/dbt/executable
MULTICELL_ACCOUNT_PREFIX=your-account-prefix
```
 You will need this file for integrating with MCP-compatible tools. 
 Here are the environment variables you could supply:

> ### Setting Environment Variables

Author: ```suggestion
### Setting environment variables
```

You will need to configure environment variables to access the tools. If you are only using the dbt CLI commands, you do not need to supply the dbt platform-specific environment variables, and vice versa. 

> #### Configuration for Discovery, Semantic Layer, and SQL Tools

Author: ```suggestion
#### Configuration for APIs and SQL tools
```

| Environment Variable | Required | Description |
| --- | --- | --- |
| DBT_HOST | Required | Your dbt Platform instance hostname. For more information, click [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses). If you are using Multi-cell, don’t include the `ACCOUNT_PREFIX` here. The default is `cloud.getdbt.com`  |
| MULTICELL_ACCOUNT_PREFIX | Only required for Multi-cell instances | Set your Multi-cell  `ACCOUNT_PREFIX`. If you are not using Multi-cell, don't set this value. You can learn more about regions and hosting [here](https://docs.getdbt.com/docs/cloud/about-cloud/access-regions-ip-addresses).  |
| DBT_TOKEN | Required | Your personal access token or service token from the dbt platform. <br/> **Note**: A service token is required when using the Semantic Layer, and this service token should have at least `Semantic Layer Only`, `Metadata Only`, and `Developer` permissions.  |
| DBT_PROD_ENV_ID | Required | Your dbt Cloud production environment ID |

> #### Additional Configuration for SQL Tools

Author: ```suggestion
#### Additional configuration for SQL tools
```

#### Disabling tools
We support disabling tool access on the local dbt-mcp.  

| Name                     | Default | Description                                                                     |
| ------------------------ | ------- | ------------------------------------------------------------------------------- |
| `DISABLE_DBT_CLI`        | `false` | Set this to `true` to disable dbt Core, dbt Cloud CLI, and dbt Fusion MCP tools |
| `DISABLE_SEMANTIC_LAYER` | `false` | Set this to `true` to disable dbt Semantic Layer MCP tools                    |
| `DISABLE_DISCOVERY`      | `false` | Set this to `true` to disable dbt Discovery API MCP tools                     |
| `DISABLE_SQL`            | `true`  | Set this to `false` to enable SQL MCP tools                                |
| `DISABLE_TOOLS`          | ""      | Set this to a list of tool names delimited by a `,` to disable specific tools    |

3. After creating your .env file, you can move on to our guides on connecting dbt-mcp to tools like Claude Desktop or Cursor or to creating a 
configuration file. This is dependent on what tools you want to integrate with. 

> ### Example Configuration File

Author: ```suggestion
### Example configuration
```
